src/map.cpp

```diff
@@ -1032,69 +1032,80 @@ void Map::removeNodeTimer(v3s16 p)
 	block->m_node_timers.remove(p_rel);
 }
 
-bool Map::isOccluded(v3s16 p0, v3s16 p1, float step, float stepfac,
-		float start_off, float end_off, u32 needed_count)
-{
-	float d0 = (float)BS * p0.getDistanceFrom(p1);
-	v3s16 u0 = p1 - p0;
-	v3f uf = v3f(u0.X, u0.Y, u0.Z) * BS;
-	uf.normalize();
-	v3f p0f = v3f(p0.X, p0.Y, p0.Z) * BS;
+bool Map::isOccluded(v3s16 pos_camera, v3s16 pos_target, float step,
+	float stepfac, float offset, float end_offset, u32 needed_count)
+{
+	float distance = BS * pos_camera.getDistanceFrom(pos_target);
+	v3f direction = intToFloat(pos_target - pos_camera, 1.0f);
+	direction.normalize();
```

Comment:
`getDistanceFrom` and `normalize` both get the length, suggestion:
```cpp
v3f direction = intToFloat(pos_target - pos_camera, 1.0f);
float distance = direction.getLength();
direction /= distance || 1.0f;
distance *= BS;
```
or
```cpp
v3f direction = intToFloat(pos_target - pos_camera, BS);
float distance = direction.getLength() * BS;
direction /= distance || 1.0f;
```
.

Reply:
Did you mean? (invalid `|| 1.0f` and double-applied `BS`)
```C++
v3f direction = intToFloat(pos_target - pos_camera, BS);
float distance = direction.getLength();
direction /= distance;
```

Reply:
Oops, the `BS` was indeed double-applied. >_<

But the `|| 1.0f` is to avoid nan for distance being `0.0f`